examples/tutorial/auto_parallel/bench_utils.py

```diff
@@ -1,10 +1,13 @@
 import time
+from copy import deepcopy
```

Comment:
As a tutorial, it should be expected that comments are almost everywhere, at least detailed comments for each section of your code, so that the user can better understand what is going on. However, I see no comments in this file. Please do this for the jupyter notebook as well.

Reply:
Sure I will add some later

Reply:
How shall we present the failure of training Resnet152 without checkpoint in `.py`?

Reply:
I think maybe in README?